docs/azure/sdk/snippets/authentication/system-assigned-managed-identity/Program.cs

```diff
@@ -0,0 +1,87 @@
+﻿using Azure.Identity;
+using Microsoft.Extensions.Azure;
+using Azure.Storage.Blobs;
+using Azure.Core;
+
+var userAssignedClientId = "<user-assigned-client-id>";
+var builder = WebApplication.CreateBuilder(args);
+
+var runningOnLocalDevBox = false;
+
+#region snippet_MIC_UseCredential
+builder.Services.AddAzureClients(clientBuilder =>
+{
+    TokenCredential credential = null;
+
+    if (runningOnLocalDevBox)
```

Comment:
Since this is an ASP.NET Core app, we have a nice way to detect the environment. Replace `runningOnLocalDevBox` with `builder.Environment.IsProduction()`.